contracts/stake-cw721/src/msg.rs

```diff
@@ -0,0 +1,111 @@
+use cosmwasm_std::Addr;
+use cosmwasm_std::Uint128;
+use cw20::Cw20ReceiveMsg;
+use cw721::Cw721ReceiveMsg;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cw_utils::Duration;
+
+pub use cw_controllers::ClaimsResponse;
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug, Clone, PartialEq)]
+pub struct InstantiateMsg {
+    // Owner can update all configs including changing the owner. This will generally be a DAO.
+    pub owner: Option<String>,
+    // Manager can update all configs except changing the owner. This will generally be an operations multisig for a DAO.
+    pub manager: Option<String>,
+    pub nft_address: String,
+    pub reward_token_address: Option<String>,
+    pub unstaking_duration: Option<Duration>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    ReceiveNft(Cw721ReceiveMsg),
+    Receive(Cw20ReceiveMsg),
+    Unstake {
+        token_id: String,
+        reward_wallet_address: Option<Addr>,
```

Comment:
When it comes in as part of a message `Addr` does not get validated. Need to use a string and then validate in the receiving method. :)
```suggestion
        reward_wallet_address: Option<String>,
```

Reply:
Done.